js/views/baseVw.js

```diff
@@ -87,23 +87,22 @@ export default class baseVw extends View {
         }
       }
 
-      const name = $field.attr('name');
+      let name = $field.attr('name');
       const isCheckboxGroup = field.type === 'checkbox' &&
         name.endsWith('[]');
+      if (isCheckboxGroup) name = name.slice(0, name.length - 2);
 
       if (name.indexOf('.') !== -1) {
         let deepVal = val;
-        let deepName = name;
 
         if (isCheckboxGroup) {
           deepVal = this._getCheckboxGroupData($formFields.filter(`[name="${name}"]`));
```

Comment:
Since name doesn't have brackets on it any more, the line above will need to have them restored so it gets the right inputs.